### PR TITLE
[WIP] Delete tenant , deprovisioning of a tenant by removing all tenant's related data.

### PR DESCRIPTION
**Pull request by cesnietor**
When deleting a tenant it will will create a job for removing the tenant's mounts `/mnt/tdisk/<tenant>` and will redeploy all hosts.

### CODE REVIEW

File: cluster/cluster.go [hunk 1]
```diff
@@ -289,6 +290,91 @@ func CreateDeploymentWithTenants(tenants []*StorageGroupTenant, sg *StorageGroup
...
+		}
+		ctx.Tenant = sgTenantResult.StorageGroupTenant.Tenant
+		// Get the credentials for a tenant
+		tenantConf, err := GetTenantConfig(tenant)
+		if err != nil {
+			ch <- err
+			return
+		}
+		// Unset Postgres notification before deleting tenant
+		err = unsetMinioConfigPostgresNotification(sgTenantResult.StorageGroupTenant, tenantConf)
```
Comment (dvaldivia) [inline cluster/cluster.go:316]: when you delete the storage, this should also get deleted right?

File: cluster/cluster.go [hunk 2]
```diff
@@ -289,6 +290,91 @@ func CreateDeploymentWithTenants(tenants []*StorageGroupTenant, sg *StorageGroup
...
+		if err != nil {
+			log.Println(err)
+			ch <- err
+			return
+		}
+		// Wait after minios restart
+		ready := isMinioReadyRetry(ctx)
```
Comment (dvaldivia) [inline cluster/cluster.go:323]: I think this is an unnecessary delay

Comment (dvaldivia) [reply cluster/cluster.go:323]: If we don't unset the bucket notification configuration, we can just keep going with the deprovisioning

**Comment by cesnietor** [comment]
Closing since it is outdated

**Pull request closed without merging**